하우스 월드컵 기능

kc7851 opened the pull request:
화면과 네비게이션 구성

kc7851 committed:
하우스 월드컵 화면

시작하면, 플레이화면, 종료화면으로 구성

kc7851 committed:
Navigation 구성

StackNavigation보다는
SwitchNavigation이 더 어울릴것 같아서 변경

kc7851 committed:
로직 구현

입력받은 array가 있을 경우 순차적으로 진행.
리팩토링 필요 함.
UI구성 및 화면효과 구현해야 함.

kc7851 committed:
UI구성 변경

불필요한 UI삭제하고 필요한 것으로만 구성

kc7851 committed:
메소드 순서 정렬

관련있는 것들끼리 묶어서 재 정렬

hannut91 commented on screens/WorldCupPlay.js:
```diff
@@ -0,0 +1,131 @@
+
+const alert = () => {
+  Alert.alert("하나를 선택해야 합니다.");
+};
```
```js
const alert = (text) => {
  Alert.alert(text);
};
```

는 어떨까요? 크게 의미는 없는 것 같지만 ㅎㅎ

kc7851 replied:
네 그렇게 바꾸고 util로 빼야겠네요.

hannut91 commented on screens/WorldCupPlay.js:
```diff
@@ -0,0 +1,131 @@
+
+  const selected = selectedNumber => {
+    setState({ ...state, selected: selectedNumber });
+  };
```
함수이름을 `select`은 어떤가요?

kc7851 replied:
네. select가 더 맞는것 같아요.

hannut91 commented on screens/WorldCupPlay.js:
```diff
@@ -0,0 +1,131 @@
+    }
+
+    if (survive.length === 2) {
+      goToFinish(selected);
```
여기서 return 안해도 문제 없나요?

kc7851 replied:
음... 저도 저기 return 넣어야하나 생각했었는데, 이상하게도 문제가 없더라구요...?

kc7851 replied:
제가 이거 navigation을 stack이 아니라 switch로 구성했는데, 그래서 그런건지 화면이 전환되면 그 이전 화면(여기서는 WorldCupPlay)이 다 초기화가 되는것 같았어요.

hannut91 replied:
아 그래요? 신기하넹 ㅎㅎ 그래도 모호하니 `return`있는게 좋을 것 같네요!

kc7851 committed:
WorldCup Navigation 이름 변경

Stack에서 Switch로 이름 변경

kc7851 committed:
alert를 util 함수로 분리

필요시 다양한 alert를 구현할 수 있음

kc7851 committed:
UI 업데이트 및 코드 리팩토링

nextRound메소드의 구현부를 여러 메소드로 분리함.

alert는 util로 분리함.

하우스를 선택하면 빨간 테두리가 표시되도록 UI를 수정함.

kc7851 committed:
코드 리팩토링

state의 key를 상단에 정의하여 다른곳에서 사용할 때 state. 을 사용하지 않도록 변경

kc7851 committed:
리팩토링

삼항연산자를 제거하고 인자로 color를 받도록 변경

hannut91 commented on screens/WorldCupPlay.js:
```diff
@@ -0,0 +1,176 @@
+    borderWidth: 5
+
+    // justifyContent: "center",
+    // alignItems: "center"
```
필요없는 주석은 지우는게 좋습니다.

kc7851 replied:
지웠습니다

hannut91 commented on screens/WorldCupPlay.js:
```diff
@@ -0,0 +1,176 @@
+      borderColor: color,
+      borderWidth: 5
+    };
+  };
```
```js
const candidateContainer = (borderColor) => {
  return {
    flex: 0.4,
    flexDirection: "row",
    width: "90%",
    backgroundColor: "#eee",
    borderColor,
    borderWidth: 5
  };
};
```

이건 어떠신가요?

kc7851 replied:
이런 문법이 아직 익숙치 않네요. 이렇게 고쳐볼게요.

hannut91 commented on screens/WorldCupStart.js:
```diff
@@ -0,0 +1,27 @@
+  );
+}
+
+const styles = StyleSheet.create({
+  container: {
+    flex: 1,
+    justifyContent: "center",
+    alignItems: "center"
+  }
+});
```
얘는 `WorldCupStart`위에 있는게 더 자연스럽습니다. 
그래야 `function WorldCupStart`와 `WorldCupStart`가 더 가까이 있습니다.

kc7851 replied:
음.. styles를 위쪽으로 옮기라는 말씀이신가요? 보통 styles는 저 위치에 쓰더라구요.

hannut91 replied:
흠 그렇군요 ㅎㅎ 제가 ReactNative를 잘 몰라서 몰랐네요. 코드만 봤을 때 위가 자연스럽다고 생각했는데 ㅎㅎ

kc7851 replied:
예제 코드들 찾아보니까 모두 저 위치에 쓰고있긴하네요, 그리고 styles도 component처럼 개별 파일로 나누어서 관리하기도 하는군요... 일단 요건 패쓰...

hannut91 commented on screens/WorldCupStart.js:
```diff
@@ -0,0 +1,27 @@
+    //TODO: 최소 2개이상일 때 가능하도록 확인하는 로직 필요.
+    //TODO: store.houses를 넘겨 줄 예정.
+    navigation.navigate("WorldCupPlay");
+  };
```
```js
const goTo = (navigation, page, options) => {
  navigation.navigate(page, options);
};
```

이렇게 만들면 다른 곳에서도 공통으로 사용할 수 있을까요?

hannut91 commented on screens/WorldCupPlay.js:
```diff
@@ -0,0 +1,176 @@
+  const round = state.round;
+  const selected = state.selected;
+  const firstColor = state.firstColor;
+  const secondColor = state.secondColor;
```
`Destructuring`을 사용하면 다음과 같이 단순화 할 수 있습니다.

```js
const { selected, firstColor, secondColor } = state;
```

See also
  - https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

kc7851 replied:
아 맞다. 이게 있었구나 ㅋㅋ 굿굿이네요

kc7851 committed:
리팩토링

불필요 주석 제거,
Destructuring

kc7851 committed:
navigation을 util로 분리

기존 navigation.navigate를 goTo로 변경

kc7851 committed:
FileSystem wrapping 작업

사진 저장, 삭제를 위한 util 생성

kc7851 committed:
기본정보 들어간 집 8개 세팅

asyncstorage로 저장하는 로직 주석처리
데모데이 끝나고 기본저장 집 삭제, 저장하는 기능 복구

kc7851 committed:
assets에 있는 image를 로드

expo-asset 의존성 추가
카메라로 직접 찍은 사진은 파일이름이 Date.now() 값이어서 총 길이가 20을 넘지않아서 파일 url이 20보다 크면 assets파일로 인식함.

hannut91 commented on utils/fileSystem.js:
```diff
@@ -0,0 +1,33 @@
+    FileSystem.documentDirectory + name
+  );
+  if (!file.exists) {
+    await FileSystem.makeDirectoryAsync(
+      FileSystem.documentDirectory + name
+    ).catch(e => {
+      console.log(e);
+    });
```
```js
try {
  await FileSystem.makeDirectoryAsync(FileSystem.documentDirectory + name);
} catch (e) => {
  console.log(e);
}
```

이렇게 쓰는게 정상적인 사용 방법입니다.

그리고 catch를 여기서 하기 보다는 `mkdir`을 호출하는 곳에서 처리하는게 더 좋아서 여기에 catch를 안쓰는게 좋을 것 같습니다. 여기서 에러를 catch 해버리면 사용하는 곳에서는 에러를 처리해줄 수 없습니다.

kc7851 replied:
음 아직 자바스크립트 예외처리 어케해야하는지 잘 모르겠네요

hannut91 replied:
음 저도 잘 모르지만 저 함수를 사용하는 입장을 잘 생각해보면 좋을 것 같아요.

hannut91 commented on utils/fileSystem.js:
```diff
@@ -0,0 +1,33 @@
+export const cleaner = async savedList => {
+  const fileList = await FileSystem.readDirectoryAsync(PHOTOS_DIR);
+  fileList.filter(file => savedList.indexOf(file) < 0)
+    .forEach(file => remove(file));
```
`fileList`보다는 `files`가 더 좋을 것 같습니다.

```js
const files = await FileSystem.readDirectoryAsync(PHOTOS_DIR);
files.filter(file => savedList.indexOf(file) < 0)
  .forEach(remove); // 이렇게 쓸 수 있습니다.
```

kc7851 replied:
헐 대박 forEach(remove) 이게 어떻게 되는거죠..?

hannut91 replied:
ㅎㅎ 원래 `forEach`안에 callback함수를 넣는건데 기존에는 익명함수를 넣으신거고 바로 함수를 넣을수도 있습니다.

hannut91 commented on utils/fileSystem.js:
```diff
@@ -0,0 +1,33 @@
+import * as FileSystem from "expo-file-system";
+
+export const PHOTOS_DIR = FileSystem.documentDirectory + "photos/";
```
`path.join()`을 쓰는 것이 좋습니다.

See also
  - https://nodejs.org/api/path.html#path_path_join_paths

kc7851 replied:
오오 좋네요.

hannut91 commented on redux/initialState.js:
```diff
@@ -89,4 +825,4 @@ export const initialState = {
     photos: []
   }
-};
+};
```
빈 줄을 추가해주세요

hannut91 commented on screens/HouseList.js:
```diff
@@ -40,6 +40,9 @@ function HouseList({ navigation }) {
   useEffect(() => {
     dispatch(fetchHouses());
   }, []);
+  // useEffect(() => {
+  //   dispatch(saveHouses(houses));
+  // }, [houses]);
```
안쓰는 것은 삭제해주세요.

kc7851 replied:
요거는 데모데이 끝나고 다시 사용할거라 일단 남겨뒀습니다.

hannut91 replied:
그렇다면 TODO를 남겨주시는게 좋을 것 같습니다.

kc7851 committed:
피드백 적용

가장 아래에 공백한칸 남기기
//TODO 남기기
forEach의 인자로 함수를 넣기

kc7851 committed:
집 사진 추가
